### PR TITLE
Move default values to config.mk

### PR DESCRIPTION
**Pull request by bvavala**
**What this PR does / why we need it**:
(this PR complements #337 )
move the definition of some default values (directory paths) to config.mk

**Which issue(s) this PR fixes**:
these default value are set in several places, this is a first pass to centralize these configuration parameters

### CODE REVIEW

File: cmake/ConfigSGX.cmake [hunk 1]
```diff
@@ -10,27 +10,27 @@ execute_process(COMMAND getconf LONG_BIT OUTPUT_VARIABLE VAR_LONG_BIT)
...
 if("${SGX_BUILD} " STREQUAL " ")
-       set(SGX_BUILD PRERELEASE)
+    message(FATAL_ERROR "SGX_BUILD: undefined environment variable")
 endif()
 
 set(SGX_SSL "$ENV{SGX_SSL}")
 if("${SGX_SSL} " STREQUAL " ")
-       set(SGX_SSL /opt/intel/sgxssl)
+    message(FATAL_ERROR "SGX_SSL: undefined environment variable")
 endif()
```
Comment (MHBauer) [inline cmake/ConfigSGX.cmake:34]: I like these changes. Making it fatal might be something to do in the toplevel file depending on where it's used.

File: cmake/ConfigSGX.cmake [hunk 2]
```diff
@@ -10,27 +10,27 @@ execute_process(COMMAND getconf LONG_BIT OUTPUT_VARIABLE VAR_LONG_BIT)
 # set them to the usual defaults.
```
Comment (MHBauer) [inline cmake/ConfigSGX.cmake:10]: needs to change as it's failing instead of defaulting.

File: cmake/ConfigSGX.cmake (hunk 1 above)
Comment (bvavala) [reply cmake/ConfigSGX.cmake:34]: I am not following. This file is supposed to be a toplevel one that should be include in cmake-based sgx builds.

File: cmake/ConfigSGX.cmake (hunk 2 above)
Comment (bvavala) [reply cmake/ConfigSGX.cmake]: missed... fixed!

File: cmake/ConfigSGX.cmake [hunk 3]
```diff
@@ -5,32 +5,31 @@
 # get arch information
 execute_process(COMMAND getconf LONG_BIT OUTPUT_VARIABLE VAR_LONG_BIT)
 
-# Set variables for common SGX variables.
-# if they are defined in the environment, take them from there, otherwise
-# set them to the usual defaults.
+# Check for the existence of common required SGX variables,
+# and fail if these are not defined.
```
Comment (g2flyer) [inline cmake/ConfigSGX.cmake:9]:
```suggestion
# and fail if these are not defined. Note global build defaults are defined `../config.mk`.
```

Comment (bvavala) [reply cmake/ConfigSGX.cmake]: yes! -- I suppose the `co-author` line is not recognized as a sign-off. Disappointing!

File: cmake/ConfigSGX.cmake (hunk 1 above)
Comment (MHBauer) [reply cmake/ConfigSGX.cmake:33]: is regular make not ever involved in the non-cmake builds?

Comment (bvavala) [reply cmake/ConfigSGX.cmake:33]: it is. Actually the build starts with the regular make, which includes `config.mk` (where the variables are defined). Then make calls cmake, which inherits these definitions. The first thing  cmake files do (in sgx builds) is to include this `ConfigSGX.cmake`, which checks for the existence of the required definitions.

**Comment by mbrandenburger** [comment]
I also think moving this little commits to PR #337 would it make easier to review.

**Comment by bvavala** [comment]
> @bvavala What is the strategy here? Should we block this PR until #337 is merged? In that case redundant commits should be cleaned out.

correct

**Comment by bvavala** [comment]
commits ported to #337

**Pull request closed without merging**